lambdas/data-migration2/src/storeErrors.ts

```diff
@@ -0,0 +1,71 @@
+import Logger from '@cumulus/logger';
+import { WriteStream } from 'node:fs';
+
+const fs = require('fs');
+const JSONStream = require('JSONStream');
+const { finished } = require('stream');
+const { promisify } = require('util');
+const { s3 } = require('@cumulus/aws-client/services');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/storeErrors' });
+
+/**
+ * Helper to create error file write stream
+ * @param {string} migrationName         - Name of migration
+ * @param {string | undefined} timestamp - Timestamp for unit testing
+ * @returns {Object}                     - Object containing error write streams and file path
+ */
+export const createErrorFileWriteStream = (migrationName: string, timestamp?: string) => {
+  const dateString = timestamp || new Date().toISOString();
+  const filepath = `${migrationName}ErrorLog-${dateString}.json`;
+  const errorFileWriteStream = fs.createWriteStream(filepath);
+  const jsonWriteStream = JSONStream.stringify();
+  jsonWriteStream.pipe(errorFileWriteStream);
+  errorFileWriteStream.write('{ "errors": \n');
```

Comment:
This is a little confusing to me. We're writing to `errorFileWriteStream` directly, even though it seems like what we want to do is write to the JSONstream, which is piped to the error file already.

I think we could avoid this if we customized the open/close options that we pass when initializing the JSONStream, like so:

```suggestion
  const jsonWriteStream = JSONStream.stringify('{"errors": [\n', '\n,\n', '\n]}\n');
  jsonWriteStream.pipe(errorFileWriteStream);
  errorFileWriteStream.write('{ "errors": \n');
```

If we do this, then `closeErrorFileWriteStream` should also be unnecessary, if I'm understanding the docs for JSONstream correctly, because the whole point of passing a custom "close" string when setting up the JSON stream is letting it handle things for you automatically

Reply:
Thanks for the feedback! I am skeptical that I can let `stringify` end the stream, so I am still making an explicit call to end the JSON stream but I removed the closeErrorFileWriteStream function. [`3e09741` (#2179)](https://github.com/nasa/cumulus/pull/2179/commits/3e09741a7febe8ee0442fa9d57d6ff5a9d2a8910)